zephyr/wrapper.c

```diff
@@ -539,7 +541,8 @@ void sys_comp_smart_amp_init(void);
  * sof-logger. This makes smex fail. Define at least one such section to
  * fix the build when sof-logger is not used.
  */
-static inline const void *smex_placeholder_f(void)
+#if CONFIG_TRACE
+static const void *smex_placeholder_f(void)
```

Comment:
Please fix the obsolete comment too. According to it this placeholder is useful when CONFIG_TRACE is _false_

In fact I'm not even sure this placeholder is still needed at all, is it? My memories are a bit vague sorry.

Reply:
@marc-hb sorry, I'm not sure I understand what you'd like to change in the comment. What becomes wrong in it after this change? Can you suggest a new comment?

Reply:
This placeholder was added so smex does not fail at build time when there is no trace, check the original commit message. But now you're removing the placeholder when CONFIG_TRACE is false. That does not seem to make sense. Maybe something changed compared to 6 months ago, not sure what.

Reply:
not following here, if there is CONFIG_TRACE=n then why do we need smex and this linker section. I would expect CONFIG_TRACE=n and still have Zephyr trace enabled (where the zephyr tooling would be used).

Reply:
> not following here, if there is CONFIG_TRACE=n then why do we need smex and this linker section.

This was done to keep the build scripts simple, see commit ca6e3c3bff38. I'm not sure why this is not required anymore but for sure the old comment does not make sense with the new `#ifdef`.

Reply:
@marc-hb I just checked out tip of SOF and Zephyr and set `CONFIG_TRACE=n` and got:
```
$WORK/zephyr-sdk/xtensa/intel_s1000/xtensa-zephyr-elf/bin/xtensa-zephyr-elf-gcc -DBUILD_VERSION=v2.7.99-3134-g7c25fd3c2e34 -DCC_OPTIMIZE_FLAGS=\"-O2\" -DKERNEL -DSOF_GIT_TAG=\"\" -DSOF_MAJOR=2 -DSOF_MICRO=0 -DSOF_MINOR=0 -DSOF_SRC_HASH=0x2317e879 -DSOF_TAG=\"2317e\" -DXCC_TOOLS_VERSION=\"zephyr\" -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -DRELATIVE_FILE=\"../modules/audio/sof/zephyr/wrapper.c\" -I$WORK/zephyr-sof/zephyr/include -I$WORK/zephyr-sof/tgl-20220111/zephyr/include/generated -I$WORK/zephyr-sof/zephyr/soc/xtensa/intel_adsp/cavs_v25 -I$WORK/zephyr-sof/zephyr/soc/xtensa/intel_adsp/cavs_v25/include -I$WORK/zephyr-sof/zephyr/soc/xtensa/intel_adsp/common/include -I$WORK/zephyr-sof/modules/hal/xtensa/include -I$WORK/zephyr-sof/modules/hal/xtensa/zephyr/soc/intel_cavs_25 -I$WORK/zephyr-sof/modules/audio/sof/zephyr/include -I$WORK/zephyr-sof/modules/audio/sof/zephyr/../src/platform/intel/cavs/include -I$WORK/zephyr-sof/modules/audio/sof/zephyr/../src/platform/tigerlake/include -I$WORK/zephyr-sof/modules/audio/sof/zephyr/../rimage/src/include -I$WORK/zephyr-sof/modules/audio/sof/zephyr/../zephyr/include -I$WORK/zephyr-sof/modules/audio/sof/zephyr/../src/include -I$WORK/zephyr-sof/modules/audio/sof/zephyr/../src/arch/xtensa/include -isystem $WORK/zephyr-sof/zephyr/lib/libc/minimal/include -isystem $WORK/zephyr-sdk/xtensa/intel_s1000/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/10.3.0/include -isystem $WORK/zephyr-sdk/xtensa/intel_s1000/xtensa-zephyr-elf/bin/../lib/gcc/xtensa-zephyr-elf/10.3.0/include-fixed -O2 -imacros $WORK/zephyr-sof/tgl-20220111/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -imacros $WORK/zephyr-sof/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fstrict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=$WORK/zephyr-sof/zephyr/samples/subsys/audio/sof=CMAKE_SOURCE_DIR -fmacro-prefix-map=$WORK/zephyr-sof/zephyr=ZEPHYR_BASE -fmacro-prefix-map=$WORK/zephyr-sof=WEST_TOPDIR -ffunction-sections -fdata-sections -mlongcalls -std=c99 -nostdinc -std=gnu99 -fno-inline-functions -MD -MT modules/sof/CMakeFiles/modules_sof.dir/wrapper.c.obj -MF modules/sof/CMakeFiles/modules_sof.dir/wrapper.c.obj.d -o modules/sof/CMakeFiles/modules_sof.dir/wrapper.c.obj -c $WORK/zephyr-sof/modules/audio/sof/zephyr/wrapper.c
$WORK/zephyr-sof/modules/audio/sof/zephyr/wrapper.c: In function 'smex_placeholder_f':
$WORK/zephyr-sof/modules/audio/sof/zephyr/wrapper.c:544:2: warning: implicit declaration of function '_DECLARE_LOG_ENTRY' [-Wimplicit-function-declaration]
  544 |  _DECLARE_LOG_ENTRY(LOG_LEVEL_DEBUG,
      |  ^~~~~~~~~~~~~~~~~~
$WORK/zephyr-sof/modules/audio/sof/zephyr/wrapper.c:548:10: error: 'log_entry' undeclared (first use in this function)
  548 |  return &log_entry;
      |          ^~~~~~~~~
$WORK/zephyr-sof/modules/audio/sof/zephyr/wrapper.c:548:10: note: each undeclared identifier is reported only once for each function it appears in
$WORK/zephyr-sof/modules/audio/sof/zephyr/wrapper.c:549:1: warning: control reaches end of non-void function [-Wreturn-type]
  549 | }
      | ^
[214/270] Building C object modules/sof/CMakeFiles/modules_sof.dir/$WORK/zephyr-sof/modules/audio/sof/src/audio/dai.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: $WORK/.local/bin/cmake --build $WORK/zephyr-sof/tgl-20220111
```
If I remove the smex place holder, I can build Zephyr with `CONFIG_TRACE=y`, but with `CONFIG_TRACE=n` I can build and sign, but smex fails. So, you're right, my patch fixes compilation but breaks smex. I didn't notice that because I wasn't running smex. Thanks for spotting! Please, check the new version.

Reply:
The new version still breaks CONFIG_TRACE=n, as predicted by the comment. Revert submitted in #5291